### PR TITLE
fix openmpi version on Hercules

### PR DESCRIPTION
**Pull request by RatkoVasic-NOAA**
### Summary

Switched from used installed openmpi 4.1.6-gcc-12.2.0-spack to system installed openmpi@4.1.4
This fixed problem on installing gcc environment on Hercules.

### Testing

Installed spack-stack-1.8.0/gcc
Wiki [page](https://github.com/JCSDA/spack-stack/wiki/spack%E2%80%90stack%E2%80%901.8.0-release-documentation) updated,

### Systems affected

Hercules

### Issue(s) addressed

#1278 

### Checklist
- [x] This PR addresses one issue/problem/enhancement, or has a very good reason for not doing so.
- [x] These changes have been tested on the affected systems and applications.
- [x] All dependency PRs/issues have been resolved and this PR can be merged.

### CODE REVIEW

**Comment by climbfuji** [comment]
That's fantastic! I guess it warrants re-tagging once more, but I am going to wait until we (hopefully) have a PR merged for S4, too.

Can you confirm that Orion is also good now? I see you updated the Wiki page for that system.

**Review by climbfuji — APPROVED** [review]

**Comment by RatkoVasic-NOAA** [comment]
> 
> Can you confirm that Orion is also good now? I see you updated the Wiki page for that system.

Confirming, both Hercules and Orion have both Intel and GNU environments installed.

**Comment by RatkoVasic-NOAA** [comment]
@climbfuji 
In the meantime I tried to run ufs-weather-model on Hercules using 1.8.0 (both Intel and GNU). It did compile, but when linking, it complained that curl is having problem with openssl. I made changes in packages.yaml to use system installed curl, openssh and openssl, and model is running now OK. ==add - it passed all 5 chosen regression tests ==

Changes:
```
<     curl:
<       externals:
<       - spec: curl@7.76.1+gssapi+ldap+nghttp2
<         prefix: /usr
100,107d62
<     openssh:
<       externals:
<       - spec: openssh@8.7p1
<         prefix: /usr
<     openssl:
<       externals:
<       - spec: openssl@3.0.1
<         prefix: /usr
```
While typing this, I saw you committed this PR. I'll do another one. Or, maybe I can try to run ufs-weather-model other machines and collect changes (if any other that Hercules) into one PR.

**Comment by climbfuji** [comment]
hmm ok - yes please, try the other platforms if you can

**Comment by RatkoVasic-NOAA** [comment]
@climbfuji here we stand with ufs-weather-model:

1. Hera, Gaea-C5, and Hercules (with curl/openssl changes) works OK.
2. Orion is missing zlib-ng (old issue #1055).  I'm installing zlib-ng right now, but it takes time, lots of dependencies. Hopefully this will work.
3. ufs-weather-model-env is missing SP library everywhere. Only Derecho is using it, so it failed. When I load manually sp lib, Derecho compiled OK.
4. Jet - failed in cmake, I'll take a look into this next.

Can you please check why is 1.8.0 skipping sp library for ufs-weather-model-env/1.0.0.lua? BTW, it is missing in 1.7.0 as well, 1.6.0 is OK.

**Comment by climbfuji** [comment]
Regarding `sp`, this is expected and the solution is to manually load it if needed. This is not an error.

See also https://github.com/JCSDA/spack-stack/issues/1330

**Comment by RatkoVasic-NOAA** [comment]
OK, we can add it in ufs-weather-model derecho module file.

**Comment by climbfuji** [comment]
I don't understand why that is needed. All UFS platforms currently load the individual modules explicitly, including `sp`.

https://github.com/ufs-community/ufs-weather-model/blob/develop/modulefiles/ufs_common.lua

**Comment by RatkoVasic-NOAA** [comment]
Not all, Derecho is (for reason unknown to me) using:
```
setenv("CMAKE_Platform", "derecho.intel")
load("ufs-weather-model-env")
```
in ufs_derecho.intel.lua

**Comment by climbfuji** [comment]
> Not all, Derecho is (for reason unknown to me) using:
> 
> ```
> setenv("CMAKE_Platform", "derecho.intel")
> load("ufs-weather-model-env")
> ```
> 
> in ufs_derecho.intel.lua

Now I remember ... but I don't remember why we had to do this. Ok, makes sense then.

**Pull request merged**
